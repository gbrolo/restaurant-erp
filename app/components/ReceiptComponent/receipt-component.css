.receipt-component-container {
    width: 100%;
}

.input-create {
    margin-bottom: 20px;
}