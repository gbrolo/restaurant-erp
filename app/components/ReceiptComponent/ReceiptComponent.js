import React, { Component } from 'react'
import { Card, CardBody, CardTitle, CardText } from "reactstrap"

import axios from 'axios'

import './receipt-component.css'

class ReceiptComponent extends Component {
    constructor(props) {
        super(props)

        this.state = {     
            receipt: this.props.receipt,
            receiptItems: null
        }
    }  
    
    componentDidMount = () => {
        const { receipt } = this.state

        let receiptItems = []

        this.axiosCancelSource = axios.CancelToken.source()

        receipt.receiptItems.forEach((item, index, array) => {
            axios({
                method: 'POST',
                url: 'http://35.166.113.228:8080/order-products/get',                    
                data: { id: item.id },
                cancelToken: this.axiosCancelSource.token
            }).then(response => {
                // console.log('response', response)
                if (response.data.code === 200 && response.data.status === 'success') {
                    const orderProduct = JSON.parse(response.data.product)
                    orderProduct['quantity'] = item.quantity
                    // console.log(orderProduct)
                    receiptItems.push(orderProduct)

                    this.setState({ receiptItems })
                }
            }).catch(error => {
                
            })                
        })
    }
    
    componentWillUnmount = () => {
        this.axiosCancelSource.cancel('Receipt unmounted.')
    }

    render = () => {     
        const { receipt, receiptItems } = this.state
        // console.log(receipt)

        return (
            <div className="receipt-component-container">
                {
                    receiptItems != null &&
                    <Card>
                        <CardBody>
                            <CardTitle><strong>Factura: </strong>{receipt.id}</CardTitle>
                            <CardText><strong>Nombre: </strong>{receipt.name}</CardText>
                            <CardText><strong>Nit: </strong>{receipt.nit}</CardText>
                            <CardText><strong>Dirección: </strong>{receipt.address}</CardText>
                            <CardText><strong>Fecha: </strong>{new Date(receipt.date_created.seconds * 1000).toString()}</CardText>
                            {
                                receiptItems.map((item, index) => {
                                    return(
                                        <CardText key={index}>({item.quantity}) {item.name}, Total: Q. {item.retailPrice * item.quantity}</CardText>
                                    )
                                })
                            }
                            <CardText><strong>Subtotal </strong>Q. {parseFloat(receipt.total)}</CardText>
                           
                            <CardText><strong>Total más impuestos: </strong>Q. {parseFloat(receipt.total) + parseFloat(receipt.total * 0.12)}</CardText>
                        </CardBody>
                    </Card>
                }
            </div>
        )
    }
}

export default ReceiptComponent