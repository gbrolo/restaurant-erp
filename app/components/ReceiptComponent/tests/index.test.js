import React from 'react'
import { render } from 'react-testing-library'
import ReceiptComponent from '../ReceiptComponent'

const receipt = {
    id: 'uRiL30MvGKG6fh9ERXSJ',
    name: 'Kafka',
    nit: '319823-9',
    address: 'Guatemala',
    date_created: {
        seconds: 100000000000
    },
    receiptItems: [
        { id: '6Wukpz6leyWxig1DPa9P', quantity: 2 },
        { id: 'JGyMjYUoi1ierSYpbZLx', quantity: 3 }
    ],
    total: 115
}

describe('<ReceiptComponent />', () => {
    it('renders the ReceiptComponent component and matches snapshot', async () => {
        const { container } = await render(
            <ReceiptComponent 
                receipt={receipt}
            />
        )

        expect(container.firstChild).toMatchSnapshot()
    })
})