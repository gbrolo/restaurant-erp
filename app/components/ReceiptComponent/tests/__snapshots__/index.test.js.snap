// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReceiptComponent /> renders the ReceiptComponent component and matches snapshot 1`] = `
<div
  class="receipt-component-container"
/>
`;
